### PR TITLE
Add arguments_specs to roles

### PR DESCRIPTION
**Pull request by GomathiselviS**
This PR adds argument_specs.yml to the roles to enable validation of arguments.
This aligns with https://github.com/ansible-collections/cloud-content-handbook/blob/main/Roles-arguments_validation.md and https://github.com/redhat-cop/automation-good-practices/tree/main/roles#argument-validation

### CODE REVIEW

**Comment by alinabuzachis** [comment]
@GomathiselviS Can you please rebase? Thanks.

**Comment by GomathiselviS** [comment]
> @GomathiselviS Can you please rebase? Thanks.

Certainly, I will mark it as a draft until the PR is ready for review.

File: changelogs/fragments/role_arg_specs.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+---
+trivial:
```
Comment (abikouo) [inline changelogs/fragments/role_arg_specs.yaml:2]: I would rather prefer having this as a `minor_changes`

File: roles/azure_load_balancer_with_public_ip/meta/argument_specs.yml [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    options:
+      azure_load_balancer_with_public_ip_operation:
+        description:
+          - Operation to perform
+        default: "create"
+        choices: ["create", "delete"]
+      azure_load_balancer_with_public_ip_resource_group:
+        description:
+          - Resource group on/from which the load balancer will reside.
+          - When `azure_load_balancer_with_public_ip_operation` is set to create, this resource group will be created if it does not exist.
```
Comment (abikouo) [inline roles/azure_load_balancer_with_public_ip/meta/argument_specs.yml:18]:
```suggestion
          - When O(azure_load_balancer_with_public_ip_operation) is set to create, this resource group will be created if it does not exist.
```

File: roles/azure_manage_network_interface/meta/argument_specs.yml [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+          - Object used to provide details for a network interface.
+        type: dict
+        options:
+          name:
+            description: Name of the network interface.
+            required: true
+          vnet_name:
+            description:
+              - Name of the existing azure virtual network where the network interface will reside.
+              - Required when `azure_manage_network_interface_operation`=create.
```
Comment (abikouo) [inline roles/azure_manage_network_interface/meta/argument_specs.yml:30]:
```suggestion
              - Required when O(azure_manage_network_interface_operation=create).
```

File: roles/azure_manage_network_interface/meta/argument_specs.yml [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+          subnet_name:
+            description:
+              - Name of the existing azure subnet where the network interface will reside.
+              - Required when `azure_manage_network_interface_operation`=create.
```
Comment (abikouo) [inline roles/azure_manage_network_interface/meta/argument_specs.yml:34]:
```suggestion
              - Required when O(azure_manage_network_interface_operation=create).
```

File: roles/azure_manage_network_interface/meta/argument_specs.yml [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+              - Name of the existing security group with which to associate the network interface.
+              - If not provided, a default security group will be created when create_with_security_group=true.
+          create_with_security_group:
+            description: Whether or not a default security group should be created with the network interface.
+            type: bool
+            default: true
+          os_type:
+            description:
+              - Determines any rules to be added to a network interface's default security group.
+              - If `os_type=Windows`, a rule allowing RDP access will be added.
```
Comment (abikouo) [inline roles/azure_manage_network_interface/meta/argument_specs.yml:46]:
```suggestion
              - If O(os_type=Windows), a rule allowing RDP access will be added.
```

File: roles/azure_manage_network_interface/meta/argument_specs.yml [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+              - If `os_type=Linux`, a rule allowing SSH access will be added.
```
Comment (abikouo) [inline roles/azure_manage_network_interface/meta/argument_specs.yml:47]:
```suggestion
              - If O(os_type=Linux), a rule allowing SSH access will be added.
```

File: roles/azure_manage_network_interface/meta/argument_specs.yml [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+            description: List of IP configurations.
+            type: list
+            elements: dict
+            options:
+              name:
+                description: Name of the IP configuration.
+                required: true
+              primary:
+                description:
+                  - Set to 'yes' to make IP configuration the primary one.
```
Comment (abikouo) [inline roles/azure_manage_network_interface/meta/argument_specs.yml:67]:
```suggestion
                  - Set to V(yes) to make IP configuration the primary one.
```

File: roles/azure_manage_network_interface/meta/argument_specs.yml [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+                  - The first IP configuration is by default set to primary=yes.
```
Comment (abikouo) [inline roles/azure_manage_network_interface/meta/argument_specs.yml:68]:
```suggestion
                  - The first IP configuration is by default set to O(primary=yes).
```

File: roles/azure_manage_postgresql/meta/argument_specs.yml [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+          - The administrator's login name of a server.
+          - Can only be specified when the server is being created (and is required for creation).
+      azure_manage_postgresql_postgresql_admin_password:
+        description:
+          - The password of the administrator login.
+          - When this is not defined, the role will generated a password that can be read later in the variable name.
+      azure_manage_postgresql_postgresql_create_mode:
+        description:
+          - Create mode of SQL Server.
+          - restore from geo redundant (geo_restore), or restore from point in time (point_in_time_restore).
```
Comment (abikouo) [inline roles/azure_manage_postgresql/meta/argument_specs.yml:80]:
```suggestion
          - restore from geo redundant V(geo_restore), or restore from point in time V(point_in_time_restore).
```

File: roles/azure_manage_postgresql/meta/argument_specs.yml [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+            description:
+              - The charset of the database. Check PostgreSQL documentation for possible values.
+              - This is only set on creation, use force to recreate a database if the values don't match.
+          collation:
+            description:
+              -The collation of the database.
+              - Check PostgreSQL documentation. This is only set on creation, use force to recreate a database if the values don't match.
+          force:
+            description:
+              - When set to `True`, will delete and recreate the existing PostgreSQL database if any of the properties don't match what is set.
```
Comment (abikouo) [inline roles/azure_manage_postgresql/meta/argument_specs.yml:133]:
```suggestion
              - When set to V(True), will delete and recreate the existing PostgreSQL database if any of the properties don't match what is set.
```

File: roles/azure_manage_resource_group/meta/argument_specs.yml [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+        description: Operation to perform.
+        choices: ["create", "delete"]
+        required: true
+      azure_manage_resource_group_name:
+        description: Resource group to create or delete.
+        required: true
+      azure_manage_resource_group_region:
+        description: An Azure location for the resource group to create.
+      azure_manage_resource_group_lock_resource_group:
+        description: If set to 'true', will lock the resource group created.
```
Comment (abikouo) [inline roles/azure_manage_resource_group/meta/argument_specs.yml:20]:
```suggestion
        description: If set to V(true), will lock the resource group created.
```

File: roles/azure_manage_security_group/meta/argument_specs.yml [hunk 12]
```diff
@@ -0,0 +1,93 @@
...
+                  - Allow to allow traffic flow.
+                  - Deny to deny traffic flow.
+                default: 'Allow'
+                choices: ['Allow', 'Deny']
+              description:
+                description: description of the rule
+              destination_address_prefix:
+                description:
+                  - The destination address prefix. CIDR or destination IP range.
+                  - Asterisk '*' can be used to match all source IPs.
```
Comment (abikouo) [inline roles/azure_manage_security_group/meta/argument_specs.yml:49]:
```suggestion
                  - Asterisk V(*) can be used to match all source IPs.
```

File: roles/azure_manage_security_group/meta/argument_specs.yml [hunk 13]
```diff
@@ -0,0 +1,93 @@
...
+                  - Default tags such as 'VirtualNetwork', 'AzureLoadBalancer' and 'Internet' can also be used.
+                  - Accepts string or a list of strings.
+                  - Asterisk '*' and default tags can only be specified as single string, not as a list of strings.
```
Comment (abikouo) [inline roles/azure_manage_security_group/meta/argument_specs.yml:52]:
```suggestion
                  - Asterisk V(*) and default tags can only be specified as single string, not as a list of strings.
```

File: roles/azure_manage_security_group/meta/argument_specs.yml [hunk 14]
```diff
@@ -0,0 +1,93 @@
...
+              destination_application_security_groups:
+                description:
+                  - List of the destination application security groups.
+                  - It could be a list of resource ids, names in same resource group, or dicts containing resource_group and name.
+                  - It is mutually exclusive with destination_address_prefix and destination_address_prefixes.
+                type: list
+              destination_port_range:
+                description:
+                  - Port or range of ports to which traffic is headed.
+                  - Can be a string or list of strings. Default is '*'.
```
Comment (abikouo) [inline roles/azure_manage_security_group/meta/argument_specs.yml:63]:
```suggestion
                  - Can be a string or list of strings. Default is V(*).
```

File: roles/azure_manage_security_group/meta/argument_specs.yml [hunk 15]
```diff
@@ -0,0 +1,93 @@
...
+                description: Indicates direction of traffic flow.
+                choices: ['Inbound', 'Outbound']
+                default: 'Inbound'
+              protocol:
+                description: Accepted traffic protocol.
+                choices: ['*', 'Udp', 'Tcp', 'Icmp']
+                default: '*'
+              source_address_prefix:
+                description:
+                  - The CIDR or source IP range. Asterisk '*' can be used to match all source IPs.
```
Comment (abikouo) [inline roles/azure_manage_security_group/meta/argument_specs.yml:75]:
```suggestion
                  - The CIDR or source IP range. Asterisk V(*) can be used to match all source IPs.
```

File: roles/azure_virtual_machine_with_public_ip/meta/argument_specs.yml [hunk 16]
```diff
@@ -0,0 +1,105 @@
...
+      - This role requires an azure user account with valid permission.
+    options:
+      azure_virtual_machine_with_public_ip_operation:
+        description:
+          - Operation to perform
+        default: "create"
+        choices: ['create', 'delete', 'power_on', 'power_off', 'deallocate', 'restart']
+      azure_virtual_machine_with_public_ip_remove_on_absent:
+        description:
+          - Specify which resources to remove when `azure_virtual_machine_with_public_ip_operation='delete'`.
```
Comment (abikouo) [inline roles/azure_virtual_machine_with_public_ip/meta/argument_specs.yml:17]:
```suggestion
          - Specify which resources to remove when O(azure_virtual_machine_with_public_ip_operation=delete).
```

File: roles/azure_virtual_machine_with_public_ip/meta/argument_specs.yml [hunk 17]
```diff
@@ -0,0 +1,105 @@
...
+          - Value 'all' removes all resources attached to the VM being removed.
+          - Value 'all_autocreated' removes the resources that were automatically created while provisioning the VM.
+          - To remove only specific resources, use the values 'network_interfaces', 'virtual_storage', or 'public_ips'.
```
Comment (abikouo) [inline roles/azure_virtual_machine_with_public_ip/meta/argument_specs.yml:20]:
```suggestion
          - To remove only specific resources, use the values V(network_interfaces), V(virtual_storage), or V(public_ips).
```

File: roles/azure_virtual_machine_with_public_ip/meta/argument_specs.yml [hunk 18]
```diff
@@ -0,0 +1,105 @@
...
+        default: 'all'
+      azure_virtual_machine_with_public_ip_resource_group:
+        description:
+          - Resource group on/from which the virtual machine will reside.
+          - When `azure_virtual_machine_with_public_ip_operation='create'`, this resource group will be created if it does not exist.
```
Comment (abikouo) [inline roles/azure_virtual_machine_with_public_ip/meta/argument_specs.yml:25]:
```suggestion
          - When O(azure_virtual_machine_with_public_ip_operation=create), this resource group will be created if it does not exist.
```

File: roles/azure_virtual_machine_with_public_ip/meta/argument_specs.yml [hunk 19]
```diff
@@ -0,0 +1,105 @@
...
+        description: Object used to provide details for a virtual machine.
+        type: dict
+        options:
+          name:
+            description: Name of the virtual machine.
+            required: true
+          admin_username:
+            description: Administrator's login name of a server. Required for creation.
+          admin_password:
+            description: Password of the administrator login. Not required when `os='Linux'` and `ssh_pw_enabled='false'`.
```
Comment (abikouo) [inline roles/azure_virtual_machine_with_public_ip/meta/argument_specs.yml:42]:
```suggestion
            description: Password of the administrator login. Not required when O(os=Linux) and O(ssh_pw_enabled=false).
```

File: roles/azure_virtual_machine_with_public_ip/meta/argument_specs.yml [hunk 20]
```diff
@@ -0,0 +1,105 @@
...
+          os:
+            description: Type of Operating System.
+            default: 'Linux'
+          availability_set:
+            description: Name or ID of existing availability set to add the VM to.
+          image:
+            description:
+              - The image used to build the VM. For custom images, the name of the image. To narrow the search to a specific resource group, a dict with the keys name and resource_group.
+              - For Marketplace images, a dict with the keys publisher, offer, sku, and version.
+              - Set `version=latest` to get the most recent version of a given image.
```
Comment (abikouo) [inline roles/azure_virtual_machine_with_public_ip/meta/argument_specs.yml:60]:
```suggestion
              - Set O(version=latest) to get the most recent version of a given image.
```

File: roles/azure_virtual_machine_with_public_ip/meta/argument_specs.yml [hunk 21]
```diff
@@ -0,0 +1,105 @@
...
+          ssh_pw_enabled:
+            description: Enable/disable SSH passwords.
+            choices: ['yes', 'no']
+            default: 'yes'
+          ssh_public_keys:
+            description:
+              - List of SSH keys when `os='Linux'`.
```
Comment (abikouo) [inline roles/azure_virtual_machine_with_public_ip/meta/argument_specs.yml:67]:
```suggestion
              - List of SSH keys when O(os=Linux).
```

**Review by abikouo — COMMENTED** [review]
This documentation format is described here https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html#semantic-markup-within-module-documentation

**Review by abikouo — APPROVED** [review]

**Review by alinabuzachis — APPROVED** [review]

**Pull request merged**